pkg/middleware/validate_action_url_test.go

```diff
@@ -0,0 +1,286 @@
+			code:                http.StatusOK,
+		},
+		{
+			name:                "POST without action header",
+			method:              "POST",
+			path:                "/api/plugins/org-generic-app",
+			actionsAllowPostURL: "",
+			addHeader:           false,
+			code:                http.StatusOK,
+		},
+		{
+			name:                "POST with action header, no paths defined",
+			method:              "POST",
+			path:                "/api/plugins/org-generic-app",
+			actionsAllowPostURL: "",
+			addHeader:           true,
+			code:                http.StatusMethodNotAllowed,
+		},
+		{
+			name:                "POST action with allowed path",
+			method:              "POST",
+			path:                "/api/plugins/org-generic-app",
+			actionsAllowPostURL: "/api/plugins/*",
+			addHeader:           true,
+			code:                http.StatusOK,
+		},
+		{
+			name:                "POST action with invalid path",
+			method:              "POST",
+			path:                "/api/notplugins/org-generic-app",
+			actionsAllowPostURL: "/api/plugins/*",
+			addHeader:           true,
+			code:                http.StatusMethodNotAllowed,
+		},
+		{
+			name:                "PUT action with valid path with header",
+			method:              "PUT",
+			path:                "/api/plugins/org-generic-app",
+			actionsAllowPostURL: "/api/plugins/*",
+			addHeader:           true,
+			code:                http.StatusOK,
+		},
+		{
+			name:                "PUT action with invalid path",
+			method:              "PUT",
+			path:                "/api/notplugins/org-generic-app",
+			actionsAllowPostURL: "/api/plugins/*",
+			addHeader:           true,
+			code:                http.StatusMethodNotAllowed,
+		},
+		{
+			name:                "PUT action with valid path without header",
+			method:              "PUT",
+			path:                "/api/plugins/org-generic-app",
+			actionsAllowPostURL: "/api/plugins/*",
+			addHeader:           false,
+			code:                http.StatusOK,
+		},
+		{
+			name:                "PUT action with invalid path without header",
+			method:              "PUT",
+			path:                "/api/notplugins/org-generic-app",
+			actionsAllowPostURL: "/api/plugins/*",
+			addHeader:           false,
+			code:                http.StatusOK,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			middlewareScenario(t, tt.name, func(t *testing.T, sc *scenarioContext) {
+				switch tt.method {
+				case "DELETE":
+					sc.m.Delete(tt.path, sc.defaultHandler)
+				case "GET":
+					sc.m.Get(tt.path, sc.defaultHandler)
+				case "HEAD":
+					sc.m.Head(tt.path, sc.defaultHandler)
+				case "OPTIONS":
+					sc.m.Options(tt.path, sc.defaultHandler)
+				case "PATCH":
+					sc.m.Patch(tt.path, sc.defaultHandler)
+				case "POST":
+					sc.m.Post(tt.path, sc.defaultHandler)
+				case "PUT":
+					sc.m.Put(tt.path, sc.defaultHandler)
+				default:
+					// POST for anything else
+					sc.m.Post(tt.path, sc.defaultHandler)
```

Comment:
We should just throw an error here, so that we're sure every test case actually has a valid method

Reply:
this should be covered now, it'll throw/handle invalid verbs